tutorial/source/tracking_1d_em.ipynb

```diff
@@ -296,23 +254,39 @@
     "        # E-step: compute soft assignments\n",
     "        with torch.no_grad():\n",
     "            assign_logits = compute_assign_logits(positions, observations, args)\n",
-    "            exists_logits = torch.empty(args.max_num_objects).fill_(\n",
-    "                math.log(args.expected_num_objects / args.max_num_objects))\n",
+    "            exists_logits=full_exists_logits[:states_loc.shape[0]]\n",
     "            assignment = MarginalAssignmentPersistent(exists_logits, assign_logits,\n",
     "                                              args.bp_iters, bp_momentum=args.bp_momentum)\n",
     "            p_exists = assignment.exists_dist.probs\n",
     "            p_assign = assignment.assign_dist.probs\n",
     "        \n",
     "        log_likelihood = detection_log_likelihood(positions,observations, args)\n",
     "        loss = -(log_likelihood * p_assign).sum()\n",
-    "        states_loc, states_cov = newton_step_2d(loss, states_loc,\n",
+    "        states_loc, _ = newton_step_2d(loss, states_loc,\n",
     "                                                args.emission_noise_scale # TODO: Play with this\n",
     "                                                )  # M-step\n",
-    "        if args.merge_threshold >= 0.0:\n",
-    "            assignment, states_loc = merge_objects(assignment, states_loc, args.merge_threshold)\n",
+    "        if args.merge_radius >= 0.0:\n",
+    "            states_loc = prune_objects(states_loc, p_exists, args.prune_threshold)\n",
+    "            states_loc = merge_objects(states_loc, args.merge_radius)\n",
     "        warn_if_nan(states_loc, 'states_loc')\n",
-    "        warn_if_nan(states_cov, 'states_cov')\n",
     "    \n",
+    "    if states_loc.shape[0] < args.max_num_objects:\n",
```

Comment:
Wait, the purpose of pruning is to remove unlikely states. Instead of adding dummy states, you should update the code to query `states_loc.shape[0]` instead of `args.max_objects`